### PR TITLE
HTCONDOR-1491 strings

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/submit_utils.cpp [hunk 1]
```diff
@@ -697,14 +697,13 @@ static void compress_path( MyString &path )
 	free( str );
 }
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/submit_utils.cpp:698]: It looks like replace_string() won't consider the replacement string as part of a new pattern to replace. So '///' would not be reduced to '/'.

File: src/condor_utils/submit_utils.cpp [hunk 2]
```diff
@@ -697,14 +697,13 @@ static void compress_path( MyString &path )
...
 /* check_path_length() has been deprecated in favor of 
  * check_and_universalize_path(), which not only checks
  * the length of the path string but also, on windows, it
  * takes any network drive paths and converts them to UNC
  * paths.
  */
-int SubmitHash::check_and_universalize_path( MyString &path )
+int SubmitHash::check_and_universalize_path( std::string &path )
 {
 	(void) path;
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/submit_utils.cpp:708]: We only check path[1] if path[0] isn't NUL. I think we're fine there.

File: src/condor_utils/submit_utils.cpp [hunk 3]
```diff
@@ -1987,16 +1985,16 @@ int SubmitHash::SetUserLog()
...
 				if (FnCheckFile) {
-					int rval = FnCheckFile(CheckFileArg, this, SFR_LOG, mulog.Value(), O_APPEND);
+					int rval = FnCheckFile(CheckFileArg, this, SFR_LOG, mulog.c_str(), O_APPEND);
 					if (rval) { ABORT_AND_RETURN( rval ); }
 				}
 
 				check_and_universalize_path(mulog);
 			}
-			AssignJobString(si->attr, mulog.Value());
+			AssignJobString(si->attr, mulog.c_str());
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/submit_utils.cpp:1997]: I will make this change.

File: src/condor_io/shared_port_client.cpp [hunk 4]
```diff
@@ -175,14 +175,14 @@ SharedPortClient::sendSharedPortID(char const *shared_port_id,Sock *sock)
...
 SharedPortClient::myName()
 {
 	// This is purely for debugging purposes.
 	// It is who we say we are when talking to the shared port server.
-	MyString name;
+	std::string name;
 	name = get_mySubSystem()->getName();
-	if( daemonCore ) {
+	if( daemonCore && daemonCore->publicNetworkIpAddr() ) {
 		name += " ";
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_io/shared_port_client.cpp:186]: It looks like there are a lot of places where we could make this change.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**